petri/src/vm/start.rs

```diff
@@ -127,7 +127,16 @@ impl PetriVmConfig {
     }
 
     /// Run the VM, launching pipette and returning a client to it.
-    pub async fn run(mut self) -> anyhow::Result<(PetriVm, PipetteClient)> {
+    pub async fn run(self) -> anyhow::Result<(PetriVm, PipetteClient)> {
+        let mut vm = self.run_with_lazy_pipette().await?;
+        let client = vm.wait_for_agent().await?;
+        Ok((vm, client))
+    }
+
+    /// Run the VM, configuring pipette to automatically start, but do not wait
+    /// for it to connect. This is useful for tests where the first boot attempt
```

Comment:
can you comment on what the behavior is if you then proceed to try and do something via petri in this case?

Reply:
It's all type system enforced. All the functions on the vm will work fine, and you don't have an agent to try to talk to pipette. Unless you call wait_for_pipette to wait for it.

Reply:
Cool.

And is `wait_for_agent` resilient against the guest rebooting?

Reply:
If you never call wait_for_agent you just have a normal vm with some unused extra disks attached.

Reply:
> And is wait_for_agent resilient against the guest rebooting?

Nope, that will fail. You need to handle the reboot in the test, then wait during the expected successful boot.

Reply:
See how our reboot tests work as an example of what this could look like.

Reply:
I think its fine to punt on the extra-disks thing (though, it might be nice to solve that as well).

> > And is wait_for_agent resilient against the guest rebooting?
> 
> Nope, that will fail. You need to handle the reboot in the test, then wait during the expected successful boot.

this is something the docs for this method should explicitly call out, given that the natural subsequent steps for folks trying to use this API would be to try and connect to petri in a lazy way

Reply:
Well the extra disks are where pipette lives, and you're asking for pipette, so they're necessary. If you don't want them you can call run_without_agent instead. And in fact you'll have to for unsupported platforms (like a BSD) since we'll fail on building the pipette disks.

As for your second comment I think we already cover it elsewhere, since we document that all petri vm functions will fail if the vm goes away for any reason and you're not expecting it, and wait_for_agent is included in that.

Reply:
I think some developer empathy is reasonable here, even if it means duping a warning that's already present in another downstream API.

Reply:
Ehhh, if you want pipette immediately you aren't calling this function. run is still public, and is going to be what's used 99% of the time.